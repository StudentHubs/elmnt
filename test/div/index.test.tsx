import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Div from '../../src/div';

describe('div', () => {

  it('no layout', () => {
    expect(renderer.create(
      <Div style={{ background: 'red' }}>
        <p>Hello</p>
        <p>World!</p>
      </Div>
    ).toJSON()).toMatchSnapshot();
  });

  it('bar layout with numeric spacing', () => {
    const tree = renderer.create(
      <Div style={{ layout: 'bar', spacing: 10, background:'red' }}>
        <div></div>
        <div></div>
        {null}
        <div></div>
      </Div>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('bar layout with string multi-spacing', () => {
    const tree = renderer.create(
      <Div style={{ layout: 'bar', spacing: '10px 20px', background:'red' }}>
        <div></div>
        <div></div>
        {null}
        <div></div>
      </Div>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('bar layout with numeric childWidths', () => {
    const tree = renderer.create(
      <Div style={{ layout: 'bar', spacing: 10, childWidths: 100, background:'red' }}>
        <div></div>
        <div></div>
        {null}
        <div></div>
      </Div>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('bar layout with string childWidths', () => {
    const tree = renderer.create(
      <Div style={{ layout: 'bar', spacing: 10, childWidths: '100px', background:'red' }}>
        <div></div>
        <div></div>
        {null}
        <div></div>
      </Div>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('bar layout with string multi-childWidths', () => {
    const tree = renderer.create(
      <Div style={{ layout: 'bar', spacing: 10, childWidths: '10px auto 20px', background:'red' }}>
        <div></div>
        <div></div>
        {null}
        <div></div>
      </Div>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('grid layout with numeric spacing', () => {
    const tree = renderer.create(
      <Div style={{ layout: 'grid', spacing: 10, background:'red' }}>
        <div></div>
        <div></div>
        {null}
        <div></div>
      </Div>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('grid layout with string spacing', () => {
    const tree = renderer.create(
      <Div style={{ layout: 'grid', spacing: '10px', background:'red' }}>
        <div></div>
        <div></div>
        {null}
        <div></div>
      </Div>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('grid layout with string multi-spacing', () => {
    const tree = renderer.create(
      <Div style={{ layout: 'grid', spacing: '10px 20px', background:'red' }}>
        <div></div>
        <div></div>
        {null}
        <div></div>
      </Div>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('stack layout with numeric spacing', () => {
    const tree = renderer.create(
      <Div style={{ layout: 'stack', spacing: 10, background:'red' }}>
        <div></div>
        <div></div>
        {null}
        <div></div>
      </Div>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('stack layout with string spacing', () => {
    const tree = renderer.create(
      <Div style={{ layout: 'stack', spacing: '10px', background:'red' }}>
        <div></div>
        <div></div>
        {null}
        <div></div>
      </Div>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('stack layout with string multi-spacing', () => {
    const tree = renderer.create(
      <Div style={{ layout: 'stack', spacing: '10px 20px', background:'red' }}>
        <div></div>
        <div></div>
        {null}
        <div></div>
      </Div>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
