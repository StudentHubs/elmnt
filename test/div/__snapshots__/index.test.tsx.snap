// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`div bar layout with child width 1`] = `
<div
  style={
    Object {
      "background": "red",
      "display": "table",
      "verticalAlign": undefined,
    }
  }
>
  <div
    style={
      Object {
        "boxSizing": "content-box",
        "display": "table-cell",
        "paddingLeft": 0,
        "verticalAlign": "middle",
        "width": 100,
      }
    }
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
        }
      }
    >
      <div
        style={
          Object {
            "width": 100,
          }
        }
      />
    </div>
  </div>
  <div
    style={
      Object {
        "boxSizing": "content-box",
        "display": "table-cell",
        "paddingLeft": 10,
        "verticalAlign": "middle",
        "width": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
        }
      }
    >
      <div />
    </div>
  </div>
  <div
    style={
      Object {
        "boxSizing": "content-box",
        "display": "table-cell",
        "paddingLeft": 10,
        "verticalAlign": "middle",
        "width": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
        }
      }
    >
      <div />
    </div>
  </div>
</div>
`;

exports[`div bar layout with multiple child width 1`] = `
<div
  style={
    Object {
      "background": "red",
      "display": "table",
      "verticalAlign": undefined,
    }
  }
>
  <div
    style={
      Object {
        "boxSizing": "content-box",
        "display": "table-cell",
        "paddingLeft": 0,
        "verticalAlign": "middle",
        "width": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
        }
      }
    >
      <div
        style={
          Object {
            "width": "10px",
          }
        }
      />
    </div>
  </div>
  <div
    style={
      Object {
        "boxSizing": "content-box",
        "display": "table-cell",
        "paddingLeft": 10,
        "verticalAlign": "middle",
        "width": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
        }
      }
    >
      <div />
    </div>
  </div>
  <div
    style={
      Object {
        "boxSizing": "content-box",
        "display": "table-cell",
        "paddingLeft": 10,
        "verticalAlign": "middle",
        "width": "20px",
      }
    }
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
        }
      }
    >
      <div
        style={
          Object {
            "width": "20px",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`div bar layout with numeric spacing 1`] = `
<div
  style={
    Object {
      "background": "red",
      "display": "table",
      "verticalAlign": undefined,
    }
  }
>
  <div
    style={
      Object {
        "boxSizing": "content-box",
        "display": "table-cell",
        "paddingLeft": 0,
        "verticalAlign": "middle",
        "width": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
        }
      }
    >
      <div />
    </div>
  </div>
  <div
    style={
      Object {
        "boxSizing": "content-box",
        "display": "table-cell",
        "paddingLeft": 10,
        "verticalAlign": "middle",
        "width": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
        }
      }
    >
      <div />
    </div>
  </div>
  <div
    style={
      Object {
        "boxSizing": "content-box",
        "display": "table-cell",
        "paddingLeft": 10,
        "verticalAlign": "middle",
        "width": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
        }
      }
    >
      <div />
    </div>
  </div>
</div>
`;

exports[`div bar layout with string multi-spacing 1`] = `
<div
  style={
    Object {
      "background": "red",
      "display": "table",
      "verticalAlign": undefined,
    }
  }
>
  <div
    style={
      Object {
        "boxSizing": "content-box",
        "display": "table-cell",
        "paddingLeft": 0,
        "verticalAlign": "middle",
        "width": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
        }
      }
    >
      <div />
    </div>
  </div>
  <div
    style={
      Object {
        "boxSizing": "content-box",
        "display": "table-cell",
        "paddingLeft": "20px",
        "verticalAlign": "middle",
        "width": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
        }
      }
    >
      <div />
    </div>
  </div>
  <div
    style={
      Object {
        "boxSizing": "content-box",
        "display": "table-cell",
        "paddingLeft": "20px",
        "verticalAlign": "middle",
        "width": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
        }
      }
    >
      <div />
    </div>
  </div>
</div>
`;

exports[`div grid layout with numeric spacing 1`] = `
<div
  style={
    Object {
      "background": "red",
    }
  }
>
  <div
    style={
      Object {
        "paddingLeft": 1,
        "paddingTop": 1,
      }
    }
  >
    <div
      style={
        Object {
          "marginLeft": "-11px",
          "marginTop": "-11px",
        }
      }
    >
      <div
        style={
          Object {
            "float": "left",
            "marginLeft": 10,
            "marginTop": 10,
            "width": "auto",
          }
        }
      >
        <div />
      </div>
      <div
        style={
          Object {
            "float": "left",
            "marginLeft": 10,
            "marginTop": 10,
            "width": "auto",
          }
        }
      >
        <div />
      </div>
      <div
        style={
          Object {
            "float": "left",
            "marginLeft": 10,
            "marginTop": 10,
            "width": "auto",
          }
        }
      >
        <div />
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "clear": "both",
        "display": "table",
      }
    }
  />
</div>
`;

exports[`div grid layout with string multi-spacing 1`] = `
<div
  style={
    Object {
      "background": "red",
    }
  }
>
  <div
    style={
      Object {
        "paddingLeft": 1,
        "paddingTop": 1,
      }
    }
  >
    <div
      style={
        Object {
          "marginLeft": "-21px",
          "marginTop": "-11px",
        }
      }
    >
      <div
        style={
          Object {
            "float": "left",
            "marginLeft": "20px",
            "marginTop": "10px",
            "width": "auto",
          }
        }
      >
        <div />
      </div>
      <div
        style={
          Object {
            "float": "left",
            "marginLeft": "20px",
            "marginTop": "10px",
            "width": "auto",
          }
        }
      >
        <div />
      </div>
      <div
        style={
          Object {
            "float": "left",
            "marginLeft": "20px",
            "marginTop": "10px",
            "width": "auto",
          }
        }
      >
        <div />
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "clear": "both",
        "display": "table",
      }
    }
  />
</div>
`;

exports[`div grid layout with string spacing 1`] = `
<div
  style={
    Object {
      "background": "red",
    }
  }
>
  <div
    style={
      Object {
        "paddingLeft": 1,
        "paddingTop": 1,
      }
    }
  >
    <div
      style={
        Object {
          "marginLeft": "-11px",
          "marginTop": "-11px",
        }
      }
    >
      <div
        style={
          Object {
            "float": "left",
            "marginLeft": "10px",
            "marginTop": "10px",
            "width": "auto",
          }
        }
      >
        <div />
      </div>
      <div
        style={
          Object {
            "float": "left",
            "marginLeft": "10px",
            "marginTop": "10px",
            "width": "auto",
          }
        }
      >
        <div />
      </div>
      <div
        style={
          Object {
            "float": "left",
            "marginLeft": "10px",
            "marginTop": "10px",
            "width": "auto",
          }
        }
      >
        <div />
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "clear": "both",
        "display": "table",
      }
    }
  />
</div>
`;

exports[`div no layout 1`] = `
<div
  style={
    Object {
      "background": "red",
    }
  }
>
  <p>
    Hello
  </p>
  <p>
    World!
  </p>
  <div
    style={
      Object {
        "clear": "both",
        "display": "table",
      }
    }
  />
</div>
`;

exports[`div stack layout with numeric spacing 1`] = `
<div
  style={
    Object {
      "background": "red",
    }
  }
>
  <div
    style={
      Object {
        "paddingTop": 0,
      }
    }
  >
    <div />
    <div
      style={
        Object {
          "clear": "both",
          "display": "table",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "paddingTop": 10,
      }
    }
  >
    <div />
    <div
      style={
        Object {
          "clear": "both",
          "display": "table",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "paddingTop": 10,
      }
    }
  >
    <div />
    <div
      style={
        Object {
          "clear": "both",
          "display": "table",
        }
      }
    />
  </div>
</div>
`;

exports[`div stack layout with string multi-spacing 1`] = `
<div
  style={
    Object {
      "background": "red",
    }
  }
>
  <div
    style={
      Object {
        "paddingTop": 0,
      }
    }
  >
    <div />
    <div
      style={
        Object {
          "clear": "both",
          "display": "table",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "paddingTop": "10px",
      }
    }
  >
    <div />
    <div
      style={
        Object {
          "clear": "both",
          "display": "table",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "paddingTop": "10px",
      }
    }
  >
    <div />
    <div
      style={
        Object {
          "clear": "both",
          "display": "table",
        }
      }
    />
  </div>
</div>
`;

exports[`div stack layout with string spacing 1`] = `
<div
  style={
    Object {
      "background": "red",
    }
  }
>
  <div
    style={
      Object {
        "paddingTop": 0,
      }
    }
  >
    <div />
    <div
      style={
        Object {
          "clear": "both",
          "display": "table",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "paddingTop": "10px",
      }
    }
  >
    <div />
    <div
      style={
        Object {
          "clear": "both",
          "display": "table",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "paddingTop": "10px",
      }
    }
  >
    <div />
    <div
      style={
        Object {
          "clear": "both",
          "display": "table",
        }
      }
    />
  </div>
</div>
`;
